tests/foreman/cli/test_katello_agent.py

```diff
@@ -46,288 +42,291 @@
-    def test_positive_register_host_ak_with_host_collection(self):
-        """Attempt to register a host using activation key with host collection
-
-        :id: 7daf4e40-3fa6-42af-b3f7-1ca1a5c9bfeb
-
-        :BZ: 1385814
-
-        :expectedresults: Host successfully registered and listed in host
-            collection
-
-        :CaseLevel: System
-        """
-        # create a new activation key
-        activation_key = make_activation_key(
-            {
-                'lifecycle-environment-id': self.env['id'],
-                'organization-id': self.org['id'],
-                'content-view-id': self.content_view['id'],
-            }
-        )
-        hc = make_host_collection({'organization-id': self.org['id']})
-        ActivationKey.add_host_collection(
-            {
-                'id': activation_key['id'],
-                'organization-id': self.org['id'],
-                'host-collection-id': hc['id'],
-            }
-        )
-        # add the registered instance host to collection
-        HostCollection.add_host(
-            {'id': hc['id'], 'organization-id': self.org['id'], 'host-ids': self.host['id']}
-        )
-        with VirtualMachine() as client:
-            client.create()
-            client.install_katello_ca()
-            # register the client host with the current activation key
-            client.register_contenthost(self.org['name'], activation_key=activation_key['name'])
-            assert client.subscribed
-            # note: when registering the host, it should be automatically added
-            # to the host collection
-            client_host = Host.info({'name': client.hostname})
-            hosts = HostCollection.hosts({'id': hc['id'], 'organization-id': self.org['id']})
-            assert len(hosts) == 2
-            expected_hosts_ids = {self.host['id'], client_host['id']}
-            hosts_ids = {host['id'] for host in hosts}
-            assert hosts_ids == expected_hosts_ids
+from robottelo.hosts import ContentHost
+
+
+@skip_if_not_set('clients', 'fake_manifest')
+@pytest.fixture(scope='module')
+def katello_agent_repos(module_ak, module_cv, module_lce, module_org):
+    """Create Org, Lifecycle Environment, Content View, Activation key"""
+    setup_org_for_a_rh_repo(
+        {
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
+            'organization-id': module_org.id,
+            'content-view-id': module_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'activationkey-id': module_ak.id,
+        }
+    )
+    # Create custom repository content
+    setup_org_for_a_custom_repo(
+        {
+            'url': FAKE_1_YUM_REPO,
+            'organization-id': module_org.id,
+            'content-view-id': module_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'activationkey-id': module_ak.id,
+        }
+    )
+    return {
```

Comment:
This isn't a required change, but I'm personally not a fan of this approach.

The tests that need these entities can simply include `module_ak`, `module_cv`, etc in their fixtures list, and reference them directly. Doing it this way turns those direct references into dictionary references dependent on the keys set here.